gio-subclass/src/application.rs

```diff
@@ -29,7 +29,20 @@ impl ArgumentList{
         }
     }
 
-    fn remove(){
+    // remove the item at index `idx` and shift the raw array
+    pub fn remove(&mut self, idx: usize){
+        self.items.remove(idx);
+
+        unsafe{
+            let n_args = glib_ffi::g_strv_length(*self.ptr);
+            assert!((idx as u32) < n_args);
+            glib_ffi::g_strfreev(*self.ptr.offset(idx as isize));
+
+            for i in (idx as u32)..n_args-1 {
+                ptr::write(self.ptr.offset(i as isize), *self.ptr.offset((i+1) as isize))
```

Comment:
You need to replace the last one with a `ptr:null()` or `ptr::null_mut()`

Reply:
Also and especially if the last item is removed (the loop won't even do one iteration in that case)